### PR TITLE
Feature cli precious

### PR DESCRIPTION
**Pull request by rylandg**
Has accompanying `precious` here https://github.com/binaris/precious/pull/212. Tests pass for my private realm but fail on testing mostly because many tests are failing on testing.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -5,7 +5,7 @@ SHELL := /bin/bash
 NO_CACHE = $(shell if [ $${NO_CACHE:-false} != false ]; then echo --no-cache; fi)
 SUDO := $(shell if docker info 2>&1 | grep "permission denied" >/dev/null; then echo "sudo -E"; fi)
 DOCKER := $(SUDO) docker
-DOCKER_IMAGE := binaris
+DOCKER_IMAGE := binaris/binaris
```
Comment (lavie) [inline Makefile]: starting to think it's time to rename this repo to `cli`. `binaris/binaris` is just too weird.

File: Makefile [hunk 2]
```diff
@@ -15,10 +15,18 @@ define cli_envs
...
+BRANCH := $(shell if [[ ! -z $${BRANCH_NAME+x} ]]; then echo $${BRANCH_NAME}; else git rev-parse --abbrev-ref HEAD 2>/dev/null || echo UNKNOWN; fi)
+
 .PHONY: build
 build: require-tag
 		$(DOCKER) build $(NO_CACHE) -f binaris.Dockerfile -t $(DOCKER_IMAGE):$(tag) .
 
+.PHONY: tag
+tag: require-tag
+		$(DOCKER) tag $(DOCKER_IMAGE):$(tag) binaris
+		$(DOCKER) tag $(DOCKER_IMAGE):$(tag) $(DOCKER_IMAGE):latest
```
Comment (lavie) [inline Makefile]: we should probably not do `latest` at all. nothing good comes out of it.

File: Jenkinsfile [hunk 3]
```diff
@@ -26,8 +31,31 @@ node {
                    make lint
                 """
             }
+            stage('Tag') {
+                echo 'Tagging local images';
+                sh """
```
Comment (lavie) [inline Jenkinsfile]: `env() { ...`?

File: Makefile (hunk 2 above)
Comment (rylandg) [reply Makefile]: https://github.com/binaris/binaris/issues/270

File: Makefile (hunk 1 above)
Comment (rylandg) [reply Makefile]: Agreed. https://github.com/binaris/binaris/issues/271

File: Jenkinsfile (hunk 3 above)
Comment (rylandg) [reply Jenkinsfile]: https://github.com/binaris/binaris/issues/272

**Pull request merged**